jujugui/static/gui/src/test/test_utils.js

```diff
@@ -1106,4 +1106,21 @@ describe('utilities', function() {
     });
   });
 
+  describe('generateCloudCredentialTag', function() {
+    let utils;
+
+    before(function(done) {
+      YUI(GlobalConfig).use('juju-view-utils', function(Y) {
+        utils = Y.namespace('juju.views.utils');
+        done();
+      });
+    });
+
+    it('can generate a cloud credential tag', function() {
+      assert.equal(
+        utils.generateCloudCredentialTag('azure', 'user-spinach', 'super-cred'),
+        'cloudcred-azure_spinach_super-cred');
+    });
```

Comment:
Can credentials have dashes in them? It took me some time to generate this string by piecing together the values from the core tests. None of them had dashes after the `cloudcred-` It seems to also fail silently if things aren't "just right"